### PR TITLE
Issue 17269 cache fs ops 2

### PR DESCRIPTION
**Pull request by fabrizzio-dotCMS**

### CODE REVIEW

**Review by jdotcms — APPROVED** [review]

**Review by jgambarios — APPROVED** [review]

File: dotCMS/src/integration-test/java/com/dotmarketing/portlets/fileassets/business/FileAssetAPITest.java [hunk 1]
```diff
@@ -168,6 +170,54 @@ public void Test_Rename_File_Asset_Then_Recover_File_Then_Expect_Match()
...
+      FileAsset asset2  = APILocator.getFileAssetAPI().find(con.getInode(), APILocator.systemUser(), false);
+      assertTrue("FileAssets should be the same Object", asset == asset2);
+      
+      
+      FileAsset asset3  = APILocator.getFileAssetAPI().fromContentlet(asset2);
+      assertTrue("FileAssets should be the same Object", asset3 == asset2);
+      
+      Contentlet asset4 = APILocator.getContentletAPI().find(con.getInode(), APILocator.systemUser(), false);
+      assertTrue("Content should not be a file asset", (asset4 instanceof FileAsset));
+      assertTrue("FileAssets should be the same Object", asset3 == asset4);
```
Comment (wezell) [inline dotCMS/src/integration-test/java/com/dotmarketing/portlets/fileassets/business/FileAssetAPITest.java]: This test looks perfect, but I am missing where the magic is happening, where we put the newly transformed fileAsset back into the contentlet cache?

**Review by wezell — CHANGES_REQUESTED** [review]

File: dotCMS/src/main/java/com/dotcms/rendering/velocity/viewtools/content/FileAssetMap.java [hunk 2]
```diff
@@ -26,28 +26,17 @@ public String getUri() throws DotDataException {
...
-        FileAssetMap fam = new FileAssetMap();
-        fam.setHost(fa.getHost());
-        fam.setBinary(FileAssetAPI.BINARY_FIELD, fa.getFileAsset());
-        BeanUtils.copyProperties(fam, fa);
-
-        return fam;
+    public static FileAssetMap of(final Contentlet c) throws Exception {
+        final FileAsset fileAsset = APILocator.getFileAssetAPI().fromContentlet(c);
+        final FileAssetMap fileAssetMap = new FileAssetMap();
+        return (FileAssetMap)FileAsset.eagerlyInitializedCopy(fileAssetMap, fileAsset);
```
[the remark this answers is not in the document]
Comment (fabrizzio-dotCMS) [reply dotCMS/src/main/java/com/dotcms/rendering/velocity/viewtools/content/FileAssetMap.java]: @wezell Please Revise the new changes I pushed. It is also looking good on Travis.

**Review by wezell — APPROVED** [review]
I like this pattern much better!

File: dotCMS/src/main/java/com/dotmarketing/portlets/fileassets/business/FileAssetAPIImpl.java [hunk 3]
```diff
@@ -262,6 +273,22 @@ public FileAsset fromContentlet(Contentlet con) throws DotStateException {
 
 	}
 
+	@CloseDBIfOpened
+	public FileAssetMap fromFileAsset(final FileAsset fileAsset) throws DotStateException {
+		if (!fileAsset.isLoaded()) {
+		    //
```
[the remark this answers is not in the document]
Comment (fabrizzio-dotCMS) [reply dotCMS/src/main/java/com/dotmarketing/portlets/fileassets/business/FileAssetAPIImpl.java]: done

File: dotCMS/src/main/java/com/dotmarketing/portlets/fileassets/business/FileAssetAPI.java [hunk 4]
```diff
@@ -67,6 +68,14 @@
 	 */
 	List<IFileAsset> fromContentletsI(List<Contentlet> cons) throws DotStateException;
 
+	/**
+	 *
```
[the remark this answers is not in the document]
Comment (fabrizzio-dotCMS) [reply dotCMS/src/main/java/com/dotmarketing/portlets/fileassets/business/FileAssetAPI.java]: doced

File: dotCMS/src/main/java/com/dotcms/util/Loadable.java [hunk 5]
```diff
@@ -0,0 +1,9 @@
+package com.dotcms.util;
+
+public interface Loadable {
```
[the remark this answers is not in the document]
Comment (fabrizzio-dotCMS) [reply dotCMS/src/main/java/com/dotcms/util/Loadable.java]: doced

**Review by jgambarios — APPROVED** [review]

**Pull request merged**